Source/Core/Core/IOS/WFS/WFSI.cpp

```diff
@@ -308,6 +332,36 @@ IPCCommandResult WFSI::IOCtl(const IOCtlRequest& request)
     WARN_LOG(IOS_WFS, "IOCTL_WFSI_DELETE_TITLE: unimplemented");
     break;
 
+  case IOCTL_WFSI_CHANGE_TITLE:
+  {
+    u64 title_id = Memory::Read_U64(request.buffer_in);
+    u16 group_id = Memory::Read_U16(request.buffer_in + 0x1C);
+
+    // TODO: Handle permissions
+    SetCurrentTitleIdAndGroupId(title_id, group_id);
+
+    // Change home directory
+    const std::string homedir_path =
+        StringFromFormat("/vol/%s/title/%s/%s/", m_device_name.c_str(),
+                         m_current_group_id_str.c_str(), m_current_title_id_str.c_str());
+    const u16 homedir_path_len = static_cast<u16>(homedir_path.size());
+    INFO_LOG(IOS_WFS, "IOCTL_WFSI_CHANGE_TITLE: %s (path_len: 0x%x)", homedir_path.c_str(),
+             homedir_path_len);
+
+    return_error_code = -3;
+    if (homedir_path_len > 0x1FD)
+      break;
+    auto* ios = IOS::HLE::GetIOS();
+    if (!ios)
```

Comment:
Can this ever happen?

Reply:
I don't think so but I'm pretty sure I've seen these checks elsewhere.

Reply:
Done.